packages/edit-post/src/components/manage-blocks-modal/manager.js

```diff
@@ -0,0 +1,193 @@
+		);
+	} );
+};
+
+/**
+ * Converts the search term into a normalized term.
+ *
+ * @param {string} term The search term to normalize.
+ *
+ * @return {string} The normalized search term.
+ */
+export const normalizeTerm = ( term ) => {
+	// Disregard diacritics.
+	//  Input: "média"
+	term = deburr( term );
+
+	// Accommodate leading slash, matching autocomplete expectations.
+	//  Input: "/media"
+	term = term.replace( /^\//, '' );
+
+	// Lowercase.
+	//  Input: "MEDIA"
+	term = term.toLowerCase();
+
+	// Strip leading and trailing whitespace.
+	//  Input: " media "
+	term = term.trim();
+
+	return term;
+};
+
+function BlockManager( {
+	instanceId,
+	categories,
+	blockTypes,
+	disabledBlockTypes,
+	enableBlockTypes,
+	disableBlockTypes,
+	search,
+	setState,
+} ) {
+	const blockItems = blockTypes.map( ( blockType ) => ( {
+		id: blockType.name,
+		icon: blockType.icon,
+		title: blockType.title,
+		category: blockType.category,
+		hasChildBlocksWithInserterSupport: false,
+		className: classnames( 'edit-post-manage-blocks-modal__block-type', {
+			'is-inactive': includes( disabledBlockTypes, blockType.name ),
+		} ),
+	} ) );
+
+	const filteredBlockItems = search ?
+		searchItems( blockItems, categories, search ) :
+		blockItems;
+
+	// Disable reason (no-autofocus): The block manager is a modal display, not
+	// one which is always visible, and one which already incurs this behavior
+	// of autoFocus via Popover's focusOnMount.
+
+	/* eslint-disable jsx-a11y/no-autofocus */
+	return (
+		<div className="edit-post-manage-blocks-modal__content">
+			<label
+				htmlFor={ `edit-post-manage-blocks-modal__search-${ instanceId }` }
+				className="screen-reader-text"
+			>
+				{ __( 'Search for a block' ) }
+			</label>
+			<input
+				id={ `edit-post-manage-blocks-modal__search-${ instanceId }` }
+				type="search"
+				placeholder={ __( 'Search for a block' ) }
+				className="edit-post-manage-blocks-modal__search"
+				value={ search }
+				onInput={
+					( event ) => setState( { search: event.target.value } )
+				}
+				autoFocus
+			/>
+			<div className="edit-post-manage-blocks-modal__results">
+				{ categories.map( ( category ) => {
+					const categoryBlockItems = filter( filteredBlockItems, {
+						category: category.slug,
+					} );
+
+					if ( ! categoryBlockItems.length ) {
+						return null;
+					}
+
+					const isAllDisabled = categoryBlockItems.every( ( blockItem ) => {
+						return disabledBlockTypes.includes( blockItem.id );
+					} );
+
+					const toggleAllDisabled = ( isToBeDisabled ) => {
+						const blockNames = map( categoryBlockItems, 'id' );
+						if ( isToBeDisabled ) {
+							disableBlockTypes( blockNames );
+						} else {
+							enableBlockTypes( blockNames );
+						}
+					};
+
+					return (
+						<PanelBody
+							key={ category.slug }
+							title={ category.title }
+							icon={ category.icon }
+						>
+							<ToggleControl
+								label={ __( 'Disable all' ) }
```

Comment:
In the way it's phrased at the moment, this probably should be updated when the state changes to `Enable all blocks`/`Disable all blocks` to better reflect how it works and give a better explanation for people who depend on VoiceOver tools.